File: BE_case05_Blog/src/main/java/com/be_case05_blog/models/Blog.java
package com.be_case05_blog.models;

import lombok.Data;

import javax.persistence.*;
import java.util.List;

@Entity
@Data
public class Blog {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id_blog")
    private Integer id_blog;

    @Column(name = "name_blog")
    private String name_blog;

    @Column(name = "content")
    private String content;

    @Column(name = "content_full", columnDefinition = "LONGTEXT")
    private String content_full;

    @Column(name = "img_blog", columnDefinition = "LONGTEXT")
    private String img_blog;

    @Column(name = "like1",columnDefinition = " INT  default 0")
    private int like1;

    @ManyToOne
    @JoinColumn(name = "category", referencedColumnName = "id_category",nullable = false)
    private Category category;

    @OneToMany

    private List<Comment> comments;



}

File: FE_case05_Blog/src/app/model/blog.ts
export class Blog{
  id_blog!: number;
  name_blog!: string;
  content!:  string;
  content_full!:  string;
  img_blog!:string;
  like1!:number;
  category!:any;
  comments!:any;

  constructor(id_blog: number, name_blog: string, content: string, content_full: string, img_blog: string, like1: number, category: any, comments: any) {
    this.id_blog = id_blog;
    this.name_blog = name_blog;
    this.content = content;
    this.content_full = content_full;
    this.img_blog = img_blog;
    this.like1 = like1;
    this.category = category;
    this.comments = comments;
  }
}






